src/riscv.h

```diff
@@ -86,8 +87,8 @@ typedef void (*riscv_mem_write_b)(struct riscv_t *rv,
                                   riscv_byte_t data);
 
 /* system instruction handlers */
-typedef void (*riscv_on_ecall)(struct riscv_t *rv);
-typedef void (*riscv_on_ebreak)(struct riscv_t *rv);
+typedef void (*riscv_on_ecall)(struct riscv_t *rv, struct rv_insn_t *ir);
```

Comment:
`riscv.h` is a public header, and there should be no reference to IR. Hide the implementation details.